piksi_tools/console/settings_view.py

```diff
@@ -568,29 +578,101 @@ def _settings_export_to_file_button_fired(self):
+                parser.write(f)
+        except IOError as e:
+            print('Unable to export settings to file due to IOError: {}'.format(e))
+
+    def _import_success(self, count):
+        print("Successfully imported {} settings from file.".format(count))
+        confirm_prompt = prompt.CallbackPrompt(
+            title="Save to device flash?",
+            actions=[prompt.close_button, prompt.ok_button],
+            callback=self._settings_save_button_fired)
+        confirm_prompt.text = "\n" \
+            "  Settings import from file complete.  Click OK to save the settings  \n" \
+            "  to the device's persistent storage.  \n"
+        confirm_prompt.run(block=False)
+
+    def _import_fail(self):
+        confirm_prompt = prompt.CallbackPrompt(
+            title="Failed to import settings from file",
+            actions=[prompt.close_button])
+        confirm_prompt.text = "\n" \
+            "  Verify that config file is not overwriting active connection settings.  \n"
+        confirm_prompt.run(block=False)
+
+    def _import_failure_section(self, section):
+        print(("Unable to import settings from file."
+               " Setting section \"{0}\" has not been sent from device.").format(section))
+
+    def _import_failure_not_found(self, section, name):
+        print(("Unable to import settings from file. Setting \"{0}\" in section \"{1}\""
+               " has not been sent from device.").format(name, section))
+
+    def _import_failure_write(self, error, section, name):
+        print(("Writing setting \"{0}\" in section \"{1}\""
+               " failed with error code {2}.").format(name, section, error))
+
+    def _write_parsed(self, parser):
+        settings_to_write = []
+        # Iterate over each setting
+        for section, settings in parser.items():
+            if not settings:
+                # Empty
+                continue
+
+            this_section = self.settings.get(section, None)
+
+            if this_section is None:
+                self._import_failure_section(section)
+                return
+
+            for name, value in settings.items():
+                this_setting = this_section.get(name, None)
+                if this_setting is None:
+                    self._import_failure_not_found(section, name)
+                    return
+
+                settings_to_write.append({'section': section,
+                                          'name': name,
+                                          'value': value})
+
+        ret = self.settings_api.write_all(settings_to_write)
+
+        success = True
+
+        for (error, section, name, value) in ret:
+            if error and error != SettingsWriteResponseCodes.SETTINGS_WR_READ_ONLY:
+                self._import_failure_write(error, section, name)
+                success = False
+            else:
+                self.settings.get(section, None).get(name, None).value = value
```

Comment:
I think this triggers redundant write through `_value_changed` -> filter it out.